src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -210,24 +211,70 @@ async def _get_search_responses(
     parsed_metadata: List[MetadataParsedQueryParam],
     limit: int,
     offset: int,
-    query: Optional["StringQuery"] = None,
+    text_query: Optional["TextQuery"] = None,
+    vector_query: Optional["VectorQuery"] = None,
     user: Optional[User] = None,
     response_statuses: Optional[List[ResponseStatusFilter]] = None,
     sort_by_query_param: Optional[Dict[str, str]] = None,
 ) -> "SearchResponses":
+    vector_settings = None
+    record = None
+
+    if vector_query:
+        if vector_query.record_id is not None:
+            record = await datasets.get_record_by_id(db, vector_query.record_id)
+            if record is None:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f"Record with id `{vector_query.record_id}` not found in dataset `{dataset.id}`.",
+                )
+            await record.awaitable_attrs.vectors
+
+        vector_settings = await datasets.get_vector_settings_by_name_and_dataset_id(
+            db, name=vector_query.name, dataset_id=dataset.id
+        )
+        if vector_settings is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=f"Vector `{vector_query.name}` not found in dataset `{dataset.id}`.",
+            )
```

Comment:
I would move this to the beginning of the if condition (after `if vector_query`). This is something that has a big potential to fail so I think it's better to put it first.

Reply:
Yes, I will take a look this code an try to refactor/better organize it